src/resdk/tables/variant.py

```diff
@@ -0,0 +1,157 @@
+
+    @property
+    @lru_cache()
+    def variants(self) -> pd.DataFrame:
+        """Return variants table as a pandas DataFrame object."""
+        return self._load_fetch(self.VARIANTS)
+
+    @property
+    @lru_cache()
+    def depth(self) -> pd.DataFrame:
+        """Return depth table as a pandas DataFrame object."""
+        return self._load_fetch(self.DEPTH)
+
+    @property
+    @lru_cache()
+    def coverage(self) -> pd.DataFrame:
+        """Return coverage table as a pandas DataFrame object."""
+        return self._load_fetch(self.COVERAGE)
+
+    def _download_qc(self) -> pd.DataFrame:
+        """Download sample QC data and transform into table."""
+        # No QC is given for variants data - return empty DataFrame
+        return pd.DataFrame()
+
+    @staticmethod
+    def construct_index(row) -> str:
+        """
+        Construct index of the variants table.
+
+        Index should have the form:
+        <chr>_<position>_<snp_change>_<aminoacid_pos_change>
+        E.g. chr2_1234567_C>T_Gly12Asp
+        """
+        chrom = row["CHROM"]
+        pos = row["POS"]
+        ref = row["REF"]
+        alt = row["ALT"]
+        aa_change = row["HGVS.p"]
+        # Often the amino acid change is given as p.Gly12Asp
+        # Remove redundant "p." prefix:
+        if aa_change.startswith("p."):
+            aa_change = aa_change[2:]
+
+        return f"{chrom}_{pos}_{ref}>{alt}_{aa_change}"
+
+    @staticmethod
+    def encode_mutation(row) -> int:
+        """Encode mutation to numerical value.
+
+        Mutations are given as <allele1>/<allele2>, e.g. T/T or C/T
+
+        Encode these mutations as:
+            - 0 for wild type (no mutation)
+            - 1 for heterozygous mutation
+            - 2 for homozygous mutation
+        """
+        try:
+            allele1, allele2 = re.match(
+                r"([ACGT])/([ACGT])", row["SAMPLENAME1.GT"]
+            ).group(1, 2)
+        except (AttributeError, TypeError):
+            # TypeError is raised when row["SAMPLENAME1.GT"] is np.nan / None
+            # AttributeError is raised when there is no match
+            return 0
+
+        if allele1 == allele2 == row["REF"]:
+            return 0
+        elif allele1 == allele2 == row["ALT"]:
+            return 2
+        else:
+            return 1
+
+    def _parse_file(self, file_obj, sample_id, data_type) -> pd.Series:
+        """Parse file - get encoded variants of single sample."""
+        sample_data = pd.read_csv(file_obj, sep="\t")
+
+        # Remove rows where ALT is NaN, e.g. no mutation:
+        sample_data.dropna(axis=0, subset=["ALT"], inplace=True)
+
+        # Construct index
+        sample_data["index"] = sample_data.apply(
+            self.construct_index, axis=1, result_type="reduce"
+        )
+        sample_data.set_index("index", inplace=True)
+
+        if data_type == self.VARIANTS:
+            s = sample_data.apply(self.encode_mutation, axis=1, result_type="reduce")
```

Comment:
This happens because there are `NaN` values present for this particluar variant. NaN's are floats, so all values in a given column are casted to float. I would rather keep the possiblity of having `np.nan`'s than to cast this to integers and present missing values with pandas.NA (which can represent missing integers).

Reply:
I agree, keeping in the `NaN` values makes sense and is more informative.